shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -244,6 +244,16 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
   [self updateEditingState];
 }
 
+- (BOOL)shouldChangeTextInRange:(UITextRange*)range replacementText:(NSString*)text {
+  if (self.returnKeyType == UIReturnKeyDone && [text isEqualToString:@"\n"]) {
```

Comment:
Tried that, but since we don't implement `UITextField`, just `UITextInput` the delegate doesn't get those. :(

I believe I may have filed a separate bug to investigate extending UITextField, which would get us this, as well as some other benefits like the positioning/selection loupes according to some WWDC materials.